src/main/java/com/moorhuhnservice/moorhuhnservice/service/ConfigService.java

```diff
@@ -0,0 +1,178 @@
+    configurationRepository.save(configuration);
+    return savedQuestionDTO;
+  }
+
+  /**
+   * Delete a question from a specific configuration
+   *
+   * @throws ResponseStatusException when configuration with the id or question with id does not exist
+   * @param id the id of the configuration where a question should be removed
+   * @param questionId the id of the question that should be deleted
+   * @return the deleted question as DTO
+   */
+  public QuestionDTO removeQuestionFromConfiguration(final UUID id, final UUID questionId) {
+    final Configuration configuration = getConfiguration(id);
+    final Optional<Question> optionalQuestion = getQuestionInConfiguration(questionId, configuration);
+    if (optionalQuestion.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Question with " + questionId + " does not exist.");
+    }
+    final Question question = optionalQuestion.get();
+    final QuestionDTO deletedQuestionDTO = questionMapper.questionToQuestionDTO(question);
+    configuration.removeQuestion(question);
+    configurationRepository.save(configuration);
+    questionRepository.delete(question);
+    return deletedQuestionDTO;
+  }
+
+  /**
+   * Update a question from a specific configuration
+   *
+   * @throws ResponseStatusException when configuration with the id or question with id does not exist
+   * @param id the id of the configuration where a question should be updated
+   * @param questionId the id of the question that should be updated
+   * @param questionDTO the content of the question that should be updated
+   * @return the updated question as DTO
+   */
+  public QuestionDTO updateQuestionFromConfiguration(
+    final UUID id,
+    final UUID questionId,
+    final QuestionDTO questionDTO
+  ) {
+    final Configuration configuration = getConfiguration(id);
+    final Optional<Question> optionalQuestion = getQuestionInConfiguration(questionId, configuration);
+    if (optionalQuestion.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Question with " + questionId + " does not exist.");
+    }
+    final Question question = questionMapper.questionDTOToQuestion(questionDTO);
+    question.setId(questionId);
+    final Question savedQuestion = questionRepository.save(question);
+    final QuestionDTO savedQuestionDTO = questionMapper.questionToQuestionDTO(savedQuestion);
+    return savedQuestionDTO;
+  }
+
+  /**
+   *
+   * @throws ResponseStatusException when question with the id in the given configuration does not exist
+   * @param questionId id of searched question
+   * @param configuration configuration in which the question is part of
+   * @return an optional of the question
+   */
+  private Optional<Question> getQuestionInConfiguration(UUID questionId, Configuration configuration) {
+    return configuration
+      .getQuestions()
+      .stream()
+      .filter(filteredQuestion -> filteredQuestion.getId().equals(questionId))
+      .findFirst();
```

Comment:
```suggestion
      .findAny();
```
Doesn't sort the elements before returning.